client/components/with-localized-moment/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import getCurrentLocaleSlug from 'state/selectors/get-current-locale-slug';
+
+const momentLoaders = new Map();
+
+function loadMomentLocale( locale ) {
+	const existingLoader = momentLoaders.get( locale );
+	if ( existingLoader ) {
+		return existingLoader;
+	}
+
+	const newLoader = import( /* webpackChunkName: "moment-locale-[request]", webpackInclude: /\.js$/ */ `moment/locale/${ locale }` )
+		.then( () => {
+			moment.locale( locale );
+		} )
+		.finally( () => {
+			momentLoaders.delete( locale );
+		} );
+	momentLoaders.set( locale, newLoader );
+	return newLoader;
+}
+
+function getDisplayName( WrappedComponent ) {
+	return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export default function( WrappedComponent ) {
+	class WithLocalizedMoment extends Component {
+		static displayName = `WithLocalizedMoment(${ getDisplayName( WrappedComponent ) })`;
+
+		loadLocale() {
+			if ( this.props.locale === moment.locale() ) {
+				return;
+			}
+			if ( this.props.locale !== 'en' ) {
+				const loadingLocale = this.props.locale;
```

Comment:
Do we need some kind of whitelist or fallback for this? This only came to mind after playing around in DevDocs and using a non-existent locale and may not be applicable to the real use case (where afaik the user chooses from a list of available locales).

Reply:
Good idea!

Reply:
Or maybe a `.catch()` which loads `en` in this case? 🤔